pkg/processor/sumologicschemaprocessor/README.md

```diff
@@ -98,3 +109,72 @@ Below is a list of all attribute keys that are being translated.
 | `k8s.statefulset.name`    | `statefulset`       |
 | `service.name`            | `service`           |
 | `log.file.path_resolved`  | `_sourceName`       |
+
+### Nesting attributes
+
+Nesting attributes allows to change the structure of attributes (both resource level and record level attributes)
+basing on their keys by nesting attributes with common paths into maps.
+Common path is defined as a common prefix that consists of strings separated by a given separator string (by default a dot - `.`)
+and end with that separator, for example: if the separator is `.`,
+then `xyz` is a common prefix path for `xyz.abc.qwe` and `xyz.foo`,
+but `sumo` **is not** a prefix path for neither `sumologic.foo.baz` nor `sumologic.bar.baz`.
+
+#### Example
+
+The following set of attributes:
+
+```json
+{
+  "kubernetes.container_name": "xyz",
+  "kubernetes.host.name": "the host",
+  "kubernetes.host.address": "127.0.0.1",
+  "kubernetes.namespace_name": "sumologic",
+  "another_attr": "42"
+}
+```
+
+Should be translated into such set:
+
+```json
+{
+  "kubernetes": {
+    "container_name": "xyz",
+    "namespace_name": "sumologic",
+    "host": {
+      "name": "the host",
+      "address": "127.0.0.1"
+    }
+  },
+ "another_attr": "42"
+}
+```
+
+#### Known issues
+
+This feature has undefined behavior when in input there are various keys that will be mapped to the same nested structure. 
+For example, given the following attributes:
+
+```json
+{
+  "a.b.c.": "d",
+  "a": {
+    "b": {
+      "c": "e"
+    }
+  }
+}
+```
+
+It is not possible to predict, what will be the result. It will have the following structure:
+
+```json
+{
+  "a": {
+    "b": {
+      "c": ...
+    }
+  }
+}
+```
+
+However, it is not possible to know a priori if the value under key `c` will be equal to `d` or `e`.
```

Comment:
There is one issue that I encountered and I think that we should definitely make it a UB.

Reply:
what is happening in current implementation?
IMHO, We shouldn't override existing keys

Reply:
In general, keys are overwritten. If there was a map, the maps are merged (for example `"a.b": "c"` and `"a": {"d":"e"}` get merged into `"a": {"b": "c", "d":"e"}`).
The problem is that we can't easily decide whether we will be overwriting a key, because it depends on the order in which `Range` method of `pcommon.Map` traverses the attributes. We would have to make some kind of preprocessing to check all existing keys.

Reply:
makes sense, I think for the first iteration it should be fine. Let's follow the KISS principle